packages/block-editor/src/components/provider/use-block-sync.js

```diff
@@ -108,6 +108,9 @@ export default function useBlockSync( {
 	// waiting for React renders for changes.
 	const onInputRef = useRef( onInput );
 	const onChangeRef = useRef( onChange );
+	onInputRef.current = onInput;
```

Comment:
This is a side effect, so it needs to be wrapped in an effect hook.

Reply:
I wrapped in an effect but I suspect that it's what's causing the issue. Subscriptions might be called with an outdated listener before the effect runs, maybe I should use useLayoutEffect. I'll need to confirm again in asblocks.

Reply:
Yes, `useLayoutEffect` is used for stuff like this that needs to be sync.

Reply:
So I tested again, and useEffect seems sufficient so i think we can move forward here.